### PR TITLE
Render literals & break up giant `renderRead` function to allow reuse of e.g. `renderExpr` across renderings

### PR DESCRIPTION
**Pull request by robmwalsh**
not a complete solution but at least fixes string rendering and should make rendering other types easier (at least shows the pattern)

partial solution to #311 

also made the rendering functions a bit more user friendly and modular

### CODE REVIEW

**Comment by jczuchnowski** [comment]
@robmwalsh this is so much needed :pray: . Could you resolve the conflicts? Unfortunately there were couple of PRs that also modified the rendering.

**Comment by robmwalsh** [comment]
really wish I did this one before the hackathon! It was interesting to see all the hacks people added to get their tests working... I broke a couple of tests which are now `@@ ignore`d but I think they need custom rendering, so can fix once #309 is merged.

btw I've noticed that `fmt` frequently doesn't do everything it needs to in one pass, so now it runs twice :) the 2nd run is < 1 second if it has nothing to do. I think that's a small price to pay for not having to do another commit/push/spin up CI.

**Comment by jczuchnowski** [comment]
No, it's not fmt, it's me :). Sometimes when I resolve conflicts when merging master to someone else's branch, I make mistakes and then the check fails. But I merge anyway, because it would take forever to contact eveyone.
It's just temporary for the hackathon tickets.
Unless it's something else you have in mind.

File: postgres/src/test/scala/zio/sql/postgresql/FunctionDefSpec.scala [hunk 1]
```diff
@@ -854,8 +872,6 @@ object FunctionDefSpec extends PostgresRunnableSpec with ShopSchema {
       } yield assert(r.head)(equalTo("UTF8"))
 
       assertion.mapErrorCause(cause => Cause.stackless(cause.untraced))
-    }
-  )
-
-  private def postgresStringEscape(s: String): String = s""" '${s}' """
+    } @@ ignore  //todo fix - select(PgClientEncoding())?
```
Comment (jczuchnowski) [inline postgres/src/test/scala/zio/sql/postgresql/FunctionDefSpec.scala:875]: `from smth` is required now. We need to rethink `Read`.

**Comment by robmwalsh** [comment]
This is a bit different, I merged master, ran `fmt`, refeshed IJ to make sure it picked up the changes, committed, and my branch failed the check. 

This stems from the fact that scalafmt isn't entirely idempotent. There are a few issues floating about but they keep closing them because it's a problem with one of their dependencies (e.g. https://github.com/scalameta/scalafmt/issues/917).  

Can we please merge this now? The ones that I broke will be resolved with the changes I requested here: https://github.com/zio/zio-sql/pull/290#discussion_r531206389

File: postgres/src/test/scala/zio/sql/postgresql/FunctionDefSpec.scala (hunk 1 above)
Comment (robmwalsh) [reply postgres/src/test/scala/zio/sql/postgresql/FunctionDefSpec.scala:875]: I raised this #323 for this one. I think we should rename `Read.Select` to `Read.SelectFrom` and `SelectBuilder` to `Read.Select`, both extending a common trait which the renderer can match on.

I think the underlying issue with this one I think is just that it needs parens and other `Function0`s don't.

This is one of the changes I was talking about in my comment on https://github.com/zio/zio-sql/pull/290#discussion_r531206389

**Review by jczuchnowski — APPROVED** [review]

**Pull request merged**